include/proxy/http2/Http2ServerSession.h

```diff
@@ -72,13 +72,13 @@ class Http2ServerSession : public PoolableSession, public Http2CommonSession
   void
   increment_current_active_connections_stat() override
   {
-    Metrics::increment(http2_rsb.current_active_server_connection_count);
+    Metrics::Gauge::increment(http2_rsb.current_active_server_connection_count);
```

Comment:
Since you had to change all of these anyway, why didn't you make increment and decrement non-static member functions:
`http2_rsb.current_active_server_connection_count->increment()`.  It's shorter, and you would get a compiler error if you tried to decrement a counter rather than a gauge.

Reply:
I thought of that, and even asked @cmcfarlen , we both preferred the static. I can change it if we think it's better. It just reads better with the static IMO.

Reply:
If you did Guage::increment() on a counter, or Counter::increment() on a guage, you would not get a compiler error.  Is this part of @cmcfarlen 's philosphy that C++ code should look like Python?